src/main/resources/google_checks.xml

```diff
@@ -368,8 +368,9 @@
     <module name="SuppressionXpathSingleFilter">
       <property name="checks" value="MethodName"/>
       <property name="query" value="//METHOD_DEF[
-                                     .//ANNOTATION/IDENT[contains(@text, 'Test')]]//*"/>
-      <property name="message" value="'[a-z][a-z0-9][_a-zA-Z0-9]*'.*"/>
+                                     ./MODIFIERS/ANNOTATION//IDENT[contains(@text, 'Test')]
+                                   ]/IDENT"/>
+      <property name="message" value="'[a-z][a-z0-9][a-zA-Z0-9]*(?:_[a-z0-9]{2}[a-zA-Z0-9]*)*'"/>
```

Comment:
Can we make it `value="'[a-z][a-z0-9][a-zA-Z0-9]*(?:_[a-z][a-z0-9][a-zA-Z0-9]*)*'"/>`
It will be clear that `[a-z][a-z0-9][a-zA-Z0-9]` is our pattern for name, and `_` is just separation between allowed names.

I was confused on why pattern so different.

Reply:
I believe that's what I had originally. 😅

Reply:
> Can we make it `value="'[a-z][a-z0-9][a-zA-Z0-9]*(?:_[a-z][a-z0-9][a-zA-Z0-9]*)*'"/>` 

This regex will not allow digits immediately after `_` ( https://github.com/checkstyle/checkstyle/pull/15617#discussion_r1744103032 ), such naming convention isn't prohibited by style guide ( https://github.com/checkstyle/checkstyle/pull/15617#discussion_r1744093928 ) so we should also not restrict it. @romani

Reply:
https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names

>Method names
Method names are written in [lowerCamelCase](https://google.github.io/styleguide/javaguide.html#s5.3-camel-case).
Method names are typically verbs or verb phrases. For example, sendMessage or stop.

Underscores may appear in JUnit test method names to separate logical components of the name, with each component written in [lowerCamelCase](https://google.github.io/styleguide/javaguide.html#s5.3-camel-case), for example transferMoney_deductsFromSource. There is no One Correct Way to name test methods.

I don't see there is definition of verbs, so words. I am ok to be it a bit restricted here and get extra feedback from users.
I don't think it is ok to name with digits from the beginning,  as style guide 

>with each component written in [lowerCamelCase](https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)

This phrase makes me think that my assumption is right. We should not allow digits to start a word/component.

Reply:
ok then, let's restrict it for now. If users demands for it then we will allow it